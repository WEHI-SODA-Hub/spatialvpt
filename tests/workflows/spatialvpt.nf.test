nextflow_workflow {

    name "Test Workflow SPATIALVPT"
    script "workflows/spatialvpt.nf"
    workflow "SPATIALVPT"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/data/example_analysis_algorithm/cellpose2_cyto2.json', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images/micron_to_mosaic_pixel_transform.csv', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/detected_transcripts.csv', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/202305010900_U2OS_small_set_VMSC00000_region_0.vzg', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cell_metadata_resegmented.csv', checkIfExists: false) ],
                    [ file('${projectDir}/tests/data/cell_by_gene_repartitioned.csv', checkIfExists: false) ],
                    [ file('${projectDir}/tests/data/cellpose_micron_space.parquet', checkIfExists: false) ],
                    []
                ])
                input[1] = 2400
                input[2] = 200
                input[3] = true
                input[4] = false
                input[5] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
