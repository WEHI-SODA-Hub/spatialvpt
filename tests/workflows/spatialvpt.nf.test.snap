{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.html:md5,78e647cb3479b2d83070a60c8be2eba0"
                    ]
                ],
                "1": [
                    "nf_core_pipeline_software_mqc_versions.yml:md5,29fec614a780248d1c561f034d0556d0"
                ],
                "report": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.html:md5,78e647cb3479b2d83070a60c8be2eba0"
                    ]
                ],
                "versions": [
                    "nf_core_pipeline_software_mqc_versions.yml:md5,29fec614a780248d1c561f034d0556d0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-26T16:28:20.231657315"
    }
}